testing/asserts.ts

```diff
@@ -411,6 +411,39 @@ export function assertNotStrictEquals(
   );
 }
 
+/**
+ * Make an assertion that `actual` and `expected` are almost equal numbers through
+ * a given epsilon. It can be used to take into account IEEE-754 double-precision
+ * floating-point representation limitations.
+ * If the values are not almost equal then throw.
+ *
+ * ```ts
+ * import { assertAlmostEquals } from "./asserts.ts";
+ *
+ * assertAlmostEquals(0.1, 0.2)
+ * ```
+ */
+export function assertAlmostEquals(
+  actual: number,
+  expected: number,
+  epsilon = 1e-7,
+  msg?: string,
+) {
+  if (actual === expected) {
+    return;
+  }
+  if (
+    (Number.isNaN(actual)) || (Number.isNaN(expected)) ||
+    (Math.abs(expected - actual) > epsilon)
+  ) {
+    const f = (n: number) =>
+      n.toFixed(-Math.log10(epsilon)).replace(/\.?0+$/, "");
+    throw new AssertionError(
+      msg ?? `actual: "${f(actual)}" expected to be close to "${f(expected)}"`,
+    );
+  }
+}
```

Comment:
> However there's one thing that I think look weird it to always force exponential notation.
> Maybe we could add a threshold, because printing `3e-1` instead of `0.3` doesn't feel natural.
> Like if less than 7 significant digits we should fallback to `toFixed()` notation ? I think it'd increase readability

True, it's just tricky to get this right. If you think the readability is worth it though, I have no objections. Just trying to err on the side of simplicity. :slightly_smiling_face:

Reply:
Yeah I think it's hard to get it right too. I'll let as is for now 😅